### DISCUSSION TITLE
smartctl --scan vs ancient 3ware RAID controllers

### DISCUSSION THREAD

**Discussion by harridu** [smartmontools/smartmontools#311] [Q&A] [answered]
Hi folks, for LSI and Dell RAID controllers and others smartctl --scan reports something like
```
/dev/bus/0 -d megaraid,8 # /dev/bus/0 [megaraid_disk_08], SCSI device
/dev/bus/0 -d megaraid,9 # /dev/bus/0 [megaraid_disk_09], SCSI device
/dev/bus/0 -d megaraid,10 # /dev/bus/0 [megaraid_disk_10], SCSI device
/dev/bus/0 -d megaraid,11 # /dev/bus/0 [megaraid_disk_11], SCSI device
/dev/bus/0 -d megaraid,13 # /dev/bus/0 [megaraid_disk_13], SCSI device
/dev/bus/0 -d megaraid,14 # /dev/bus/0 [megaraid_disk_14], SCSI device
/dev/bus/0 -d megaraid,15 # /dev/bus/0 [megaraid_disk_15], SCSI device
```
but for my ancient **3ware controllers** (eg 9750-8i) there seems to be no working scan. I have to find out the required params and add them to /etc/smartd.conf. For this to work I have to drop the DEVICESCAN line, ie I have to list **all** devices, even though smartctl would be smart enough to find them. Thats pretty painful.

Looking at the git log there is no indication that code about 3ware controllers had been removed, except for some man page fixes in 2017. Excuse me if I am too blind to see.

I understand that the 3ware controllers are hard to find today, but they do their job pretty well, and I've got quite a number in old systems, ie hosts which should be monitored closely. Would it be possible to find the ancient 3ware controllers during a scan as well?

smartmontools is version 7.4, as included in Debian.

**Comment by chrfranke** [comment, marked as answer]
Code for RAID controller support is always platform and controller specific. It always requires real hardware for testing as no VM emulates specific RAID controllers.

Historically, 3ware was the very first RAID controller support added to smartctl ([r842](https://www.smartmontools.org/browser/trunk/sm5/escalade.c?rev=842)). The code is mostly unchanged since we moved to C++ 16+ years ago ([r2623](https://www.smartmontools.org/changeset/2623)). I'm glad that it apparently still works.

If a volunteer with access to related test hardware would provide a reasonably tested patch and would be willing to maintain this part of the code for quite some time, I would consider to include this feature.
